src/metamorpho/MetamorphoSnippets.sol

```diff
@@ -78,18 +77,27 @@ contract MetaMorphoSnippets {
         return withdrawQueueList;
     }
 
-    /// @notice Returns the supply cap of a market on a MetaMorpho `vault`.
+    /// @notice Returns the sum of the supply caps of markets with the same collateral `token` on a MetaMorpho `vault`.
+    /// @dev This is a way to visualize exposure to this token.
     /// @param vault The address of the MetaMorpho vault.
-    /// @param marketParams The morpho blue market.
-    function capMarket(address vault, MarketParams memory marketParams) public view returns (uint192 cap) {
-        Id id = marketParams.id();
-        cap = IMetaMorpho(vault).config(id).cap;
+    /// @param token The collateral token.
+    function totalCapAsset(address vault, address token) public view returns (uint192 totalCap) {
+        uint256 queueLength = IMetaMorpho(vault).withdrawQueueLength();
+
+        for (uint256 i; i < queueLength; ++i) {
+            Id idMarket = IMetaMorpho(vault).withdrawQueue(i);
+            MarketParams memory marketParams = morpho.idToMarketParams(idMarket);
```

Comment:
if we want to save gas we could even display the usage of `extSloads` and only load the collateral token :D